scripts/update_lock.py

```diff
@@ -0,0 +1,206 @@
+    dependencies and optional-dependencies in the provided pyproject metadata."""
+
+    if not "project" in pyproject:
+        return pyproject
+
+    modified_pyproject = deepcopy(pyproject)
+
+    project_metadata = modified_pyproject["project"]
+
+    package_name = project_metadata.get("name")
+
+    if not package_name:
+        raise ValueError("The provided project metadata does not contain a name.")
+
+    if "dependencies" in project_metadata:
+        project_metadata["dependencies"] = exclude_from_dependency_list(
+            package_name=package_name, dependencies=project_metadata["dependencies"]
+        )
+
+    if "optional-dependencies" in project_metadata:
+        for group in project_metadata["optional-dependencies"]:
+            project_metadata["optional-dependencies"][
+                group
+            ] = exclude_from_dependency_list(
+                package_name=package_name,
+                dependencies=project_metadata["optional-dependencies"][group],
+            )
+
+    return modified_pyproject
+
+
+def fix_temp_dir_comments(file_path: Path):
+    """Fix the temp_dir comments so they don't cause a noisy diff
+
+    This will leave the top compile message intact as a point of sanity to verify that
+    the requirements are indeed being generated if nothing else changes.
+    """
+
+    with open(file_path, "r", encoding="utf-8") as file:
+        lines = file.readlines()
+
+    with open(file_path, "w", encoding="utf-8") as file:
+        for line in lines:
+            # Remove random temp directory name
+            line = re.sub(
+                r"\([^\)\(]*?pyproject\.toml\)",
+                "(pyproject.toml)",
+                line,
+            )
+            file.write(line)
+
+
+def compile_lock_file(
+    sources: list[Path],
+    output: Path,
+    upgrade: bool,
+    extras: bool,
+) -> None:
+    """From the specified sources compile a lock file using pip-compile from pip-tools
+    and write it to the specified output location.
+    """
+
+    print(f"Updating file at '{output}'...")
+
+    command = [
+        "pip-compile",
+        "--rebuild",
+        "--generate-hashes",
+        "--annotate",
+    ]
+
+    if upgrade:
+        command.append("--upgrade")
+
+    if extras:
+        command.append("--all-extras")
+
+    command += [
+        "--output-file",
+        str(output.absolute()),
+    ] + [str(source.absolute()) for source in sources]
+
+    with subprocess.Popen(
+        args=command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    ) as process:
+        stdout_data, _ = process.communicate()
+        if process.wait() != 0:
+            log = stdout_data.decode("utf-8") if stdout_data else "no log available."
+            raise RuntimeError(f"Failed to compile lock file:\n{log}")
+
+    fix_temp_dir_comments(output.absolute())
+
+
+def main(upgrade: bool = False):
+    """Update the dependency lock files located at 'requirements.txt' and
```

Comment:
Oh, by the way, I realized that it would be good to have a --check flag that only reports if changes to the lock file are required according to the .in and pyproject.toml. This --check option could then be automatically executed as part of a GitHub action. What do you think?

Reply:
At least, this is the paradigm we are following with the other update scripts.

Reply:
Added in [0946267](https://github.com/ghga-de/microservice-repository-template/pull/160/commits/094626728c823a2312ef0a00cba8d6b8a744a3d2)